src/complex/Utilities/OStreamUtilities.hpp

```diff
@@ -0,0 +1,765 @@
+
+namespace complex
+{
+namespace OStreamUtilities
+{
+
+/**
+ * @brief This is the primary datastructure used in print string assembly. It is a wrapper struct for
+ * adding additionaly functionality/readability (underlying datastructure is a std::vector)
+ * @param numrows The number of rows for the matrix
+ * @param numcols The number of columns for the matrix
+ */
+struct PrintMatrix2D
+{
+  // visual representation of string matrix
+  /*
+  * ****NOTE functionality for storing neighborlist below has been removed to preserve simplicity of use****
+  *
+    ID | DataArray1 | StringArray | DataArray2  //Note the ID column is optional this is just an example (**view note next to neightborlist**)
+  --------------------------------------------
+  | "1"   "value"      "String"      "value" |  In this example matrix, "value" represents any numeric type
+  | "2"   "value"      "String"      "value" |  in string representation based off parent dataArray typing.
+  | "3"   "value"      "String"      "value" |  [ie if DataArray1 was of type float64 then respective column
+  | "4"   "value"      "String"      "value" |  will be string represtentation of doubles to the end of array's index]
+  | "5"   "value"      "String"      "value" |  However, its important to note that these columns are not explicitly
+  | "6"   "value"      "String"      "value" |  numerically typed, so if you pull index of column below end of
+  | ...     ...          ...           ...   |  dataArray it can give odd values such as UNINITIALIZED or unexpected values(if stored below).
+
+  NeighborList Representation (must be processed horizontally)
+    ID   |  NumNeighbor  |       Values           ...        //Just an example usage
+  -----------------------------------------------------------
+  | "1"         "2"              "num"           "num"      |  In this case, UNITIALIZED are essentially empty values and will be ignored in
+  | "2"         "1"              "num"      "UNINITAILIZED" |  print if isBalanced bool is false (default)
+  | "3"         "0"         "UNINITAILIZED" "UNINITAILIZED" |  **checkBalanceType() should be run for uniform component arrays to speed up
+  | "4"         "2"              "num"           "num"      |  string assembly, but is not required [Neighbor lists and dataGroups will
+  | ...          ...              ...             ...       |  [Neighbor lists and dataGroups will almost always be unbalanced so its not
+  -----------------------------------------------------------  worth preparsing time, unless you know component count is uniform(Attribute Matrix)]**
+  */
+private:
+  size_t rows, columns;
+  // since its being allocated to heap anyways no significant advantage to array over vector (small sizing difference)
+  std::vector<std::string> matrix;
+  bool isBalanced = false;
+
+public:
+  PrintMatrix2D(size_t numrows, size_t numcols)
+  : rows(numrows)
+  , columns(numcols)
+  , matrix(numrows * numcols, std::string("UNINITIALIZED"))
```

Comment:
```suggestion
  , matrix(numrows * numcols, "UNINITIALIZED")
```
The `std::string` isn't required, and we can directly pass in the string literal.